### PR TITLE
Add mooncake in ADAL test app. Pass authority to resultFragment

### PR DESCRIPTION
**Pull request by rpdome**

### CODE REVIEW

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java [hunk 1]
```diff
@@ -47,11 +42,8 @@ public String getText() {
 
     enum DataProfile {
         GRAPH("https://graph.windows.net"),
-        SHAREPOINT_MS_DEV("https://msdevex-my.sharepoint.com"),
-        SHAREPOINT("00000003-0000-0ff1-ce00-000000000000"),
-        OFFICE_ONEDRIVE("6a9b9266-8161-4a7b-913a-a9eda19da220"),
-        SIMPLE("00000002-0000-0000-c000-000000000000");
```
Comment (shahzaibj) [inline userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:53]: Why are these removed?

Comment (shahzaibj) [inline userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:53]: Why are these removed?

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java [hunk 2]
```diff
@@ -67,15 +59,7 @@ public String getText() {
...
         ADFSV4("4b0db8c2-9f26-4417-8bde-3f0e3656f8e0"),
-        MOONCAKE("920cd637-4ca3-496b-8a6a-f7c6ca1b2b82"),
-        ONEDRIVE("af124e86-4e96-495a-b70a-90f90ab96707"),
-        OFFICE("d3590ed6-52b3-4102-aeff-aad2292ab01c"),
-        APPCHECK2_BF("f5d01c1c-abe6-4207-ae2d-5bc9af251724"),
-        GUESTCLIENT("ea5c8087-2476-489c-ae03-ad44a2ac399d"),
-        ADFSV3("68a10fc3-ead9-41b8-ac5e-5b78af044736"),
-        ADFSV2("e3fbd64e-3bdc-4fe9-b531-31660912b944"),
-        PING("6b748729-d940-4482-8724-5eb87a817a10"),
-        SHIBBOLETH("d518483c-c15b-4a00-9f59-cff3ffc1077b");
```
Comment (shahzaibj) [inline userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:78]: Why are these removed?

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java [hunk 3]
```diff
@@ -26,12 +26,7 @@
 public class Constants {
     enum AuthorityType {
         AAD_COMMON("https://login.microsoftonline.com/common"),
-        MSID_LAB2("https://login.microsoftonline.com/msidlab2.com"),
-        MSID_LAB3("https://login.microsoftonline.com/msidlab3.com"),
-        MSID_LAB4("https://login.microsoftonline.com/msidlab4.com"),
-        MSID_BLACKFOREST("https://login.microsoftonline.de/blfmsidlab1.onmicrosoft.de"),
-        MSID_FAIRFAX("https://login.microsoftonline.us/ffxmsidlab1.onmicrosoft.com"),
-        MSID_ARLINGTON("https://login.microsoftonline.us/arlmsidlab1.onmicrosoft.us"),
```
[the remark this answers is not in the document]
Comment (rpdome) [reply userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:34]: I'm not seeing them being used in any test cases.

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java (hunk 2 above)
Comment (rpdome) [reply userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:78]: I'm not seeing them being used in any test cases.

Comment (shahzaibj) [reply userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:78]: I know we try to use the `OFFICE` client id for testing some 1st party scenarios. I'm not inclined to removing these as we may end up needing them in the future.

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (shahzaibj) [reply userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:34]: I've used at least the lab ones for testing guest scenarios. I'm not inclined to removing these as we may end up needing them in the future.

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java (hunk 1 above)
Comment (shahzaibj) [reply userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java:53]: I know for sure that `SHAREPOINT` and `SIMPLE` are used in a few different tests. I'm not inclined to removing these as we may end up needing them in the future.

File: userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java (hunk 2 above)
Comment (rpdome) [reply userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/Constants.java]: Added back.
